File: langchain/js/chapter01/j-methods.js
import { ChatOpenAI } from '@langchain/openai';

const model = new ChatOpenAI("gpt-4o-mini")

const response = await model.invoke('Hi there!');
console.log(response.content);
// Hello! How can I assist you today?

const completions = await model.batch(['Hi there!', 'Bye!']);
console.log(completions.map(completion => completion.content));
// [ 'Hello! How can I assist you today?', 'Goodbye! Have a great day!' ]

for await (const token of await model.stream('Bye!')) {
    console.log(token.content);
    // Good
    // bye
    // !
    // Have
    // a
    // great
    // day
    // !
}

File: langchain/python/chapter01/j-methods.py
from langchain_openai.chat_models import ChatOpenAI

model = ChatOpenAI(model="gpt-4o-mini")

completion = model.invoke("Hi there!")
print(completion.content)
# Hello! How can I assist you today?

completions = model.batch(["Hi there!", "Bye!"])
print([c.content for c in completions])
# ['Hello! How can I assist you today?', 'Goodbye! If you have more questions in the future, feel free to ask. Take care!']

for token in model.stream("Bye!"):
    print(token.content)
    # Good
    # bye
    # !
    #  If
    #  you
    #  have
    #  any
    #  more
    #  questions
    #  in
    #  the
    #  future
    # ,
    #  feel
    #  free
    #  to
    #  ask
    # .
    #  Take
    #  care
    # !